src/intree.js

```diff
@@ -211,11 +211,15 @@ compileTasks = {
   },
   1: (config, cfg, payload) => {
     if (config.tasks.length > 0) {
+      const groupId = slugid.nice();
       config.tasks = config.tasks.map((task) => {
         if (!task.taskId) { throw Error('The taskId is absent.'); }
         return {
           taskId: task.taskId,
-          task: _.extend(task, {schedulerId: cfg.taskcluster.schedulerId}),
+          task: _.omit(_.extend(task, {
+            taskGroupId: task.taskGroupId || groupId,
+            schedulerId: cfg.taskcluster.schedulerId,
+          }), 'taskId'),
```

Comment:
This bit looks good!

Reply:
is it so that we don't have an object with `null` properties?